tools/validate_spec.py

```diff
@@ -161,6 +215,112 @@ def _validate_message(
     )
 
 
+class CoverageInformation:
+    def __init__(self, packages: List[Package], coverage: bool) -> None:
+        self.__total_message_coverage: Dict[ID, Dict[Link, bool]] = {}
+        self.__spec_files: Dict[str, List[ID]] = defaultdict(lambda: [])
+        self.__coverage = coverage
+
+        if self.__coverage:
+            for package in packages:
+                for message in package:
+                    assert isinstance(message, MessageValue)
+                    self.__total_message_coverage[message.identifier] = {
+                        link: False for link in message.model.structure
+                    }
+
+                    assert message.model.location is not None
+                    assert message.model.location.source is not None
+                    file_name = message.model.location.source.name
+                    self.__spec_files[file_name].append(message.identifier)
```

Comment:
It is usually better to do essential checks very early in a function and just return:

```Python
if not self.__coverage:
    return
```

Reply:
Done in e020204d